exonum/src/runtime/rust/macros.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Macros useful for developing `rust` services.
+
+/// Skips the invocation of the service hook for the genesis block.
+/// This macro is intended to be used within `rust` service hooks.
+#[macro_export]
+macro_rules! skip_for_genesis {
```

Comment:
Bikeshedding: I don't think that encapsulating this check into a macro is warranted. It could make more sense to introduce a checking method for `BlockchainData` or `CallContext`, like this:

```rust
fn after_transactions(&self, ctx: CallContext<'_>) -> Result<(), ExecutionError> {
    if ctx.in_genesis_block() {
        return Ok(());
    }
}
```